src/responder/nss/nss_enum.c

```diff
@@ -280,7 +280,12 @@ nss_setnetgrent_set_timeout(struct tevent_context *ev,
     struct timeval tv;
     uint32_t timeout;
 
-    timeout = enum_ctx->result[0]->domain->netgroup_timeout;
+    if (nss_ctx->cache_refresh_percent) {
+        timeout = enum_ctx->result[0]->domain->netgroup_timeout *
+            (nss_ctx->cache_refresh_percent / 100.0);
+    } else {
+        timeout = enum_ctx->result[0]->domain->netgroup_timeout;
+    }
```

Comment:
Checking the commit that removed the old code (075d8988) you can see:
```
- 	 if (step_ctx->nctx->cache_refresh_percent) {
-            lifetime = dom->netgroup_timeout *
-                (step_ctx->nctx->cache_refresh_percent / 100.0);
-        } else {
-            lifetime = dom->netgroup_timeout;
-        }
-        if (lifetime < 10) lifetime = 10;
-  	 set_netgr_lifetime(lifetime, step_ctx, netgr);
```

Based on the code that was removed, I'm wondering whether we should also have the minimum timeout always set to 10, as done before.

I'm not adding the "Changes Requested" label as this is more a question than a request to change the current code.

Reply:
Can you explain a technical reason for limiting minimal timeout? It would just make testing of feature much slower due to unnecessary sleeps.

Reply:
I'm exactly asking if there's a technical reason for limiting minimal timeout, as it was in the code **before**.

Please, do **not** bulldoze my questions and let's wait for the answer.

Reply:
I'm sorry but setting minimal timeout is not really related to this PR. It can be discussed in different ticket.

@pbrezina, you did a lot of changes as part of cache_req refactoring. What do you think? Should it block fixing regression before 1.16.1?

Reply:
@lslebodn, please, I understand that you have a **lot** of issues with me. But the least I would expect is that you could read my questions (which were not even done to you) without your vendetta hat.

I have asked whether the timeout is needed or not. I am **not** blocking a release, I am **not** the one blocking someone's else work (as you have done a **lot** while you were part of the team).

So, I just want to understand what I asked and nothing else.

Reply:
The question is imho related to this PR since it relates to part of code which was removed and is being brought back by Jakub's patches. Fabiano raised a valid and good question and a nice catch.

I think that actually having this ten seconds timeout limit was a bug. It's purpose seems to be to limit midpoint refresh case so it is not too small but it effectively disabled `entry_cache_netgroup_timeout = 0`. It should be moved to the midpoint refresh case of the if branch.

Reply:
There are few places in sssd which does not allow value of options which are lower then  hardcoded minimal value and it is not clear why such value was used.
  https://pagure.io/SSSD/sssd/blob/master/f/src/providers/ad/ad_dyndns.c#_183
  https://pagure.io/SSSD/sssd/blob/master/f/src/responder/common/responder_common.c#_1186
  https://pagure.io/SSSD/sssd/blob/master/f/src/responder/common/responder_common.c#_1576

This is the same with
```
-        if (lifetime < 10) lifetime = 10;
-  	 set_netgr_lifetime(lifetime, step_ctx, netgr);
```

In such situations, we should just trigger huge warning to logs/syslog rather then setting minimal value.  


> I think that actually having this ten seconds timeout limit was a bug. It's purpose seems to be to limit midpoint refresh case so it is not too small but it effectively disabled entry_cache_netgroup_timeout = 0. It should be moved to the midpoint refresh case of the if branch.

`entry_cache_netgroup_timeout = 0` is another corner case which semantically does not make a sense as a combination with `entry_cache_nowait_percentage`. (and thus not really related to PR). The first one says that entry should be always returned from backend and the second one says that you want to always return entry from responder + periodically refresh entries rather then waiting for expired entry.

Current version of patch is much better solution then current state (regression introduced in 1.15.0). If you can see areas which can be improved in 1.16.2 then it should be better to file a ticketrather then having regression in yet another release.

Reply:
> entry_cache_netgroup_timeout = 0 is another corner case which semantically does not make a sense as a combination with entry_cache_nowait_percentage. (and thus not really related to PR). The first one says that entry should be always returned from backend and the second one says that you want to always return entry from responder + periodically refresh entries rather then waiting for expired entry.

If the option was set to 0 then you got 10 seconds lifetime of the netgroup regardless of the midpoint refresh setting, therefore it was not possible to have the netgroup always refreshed. 

If the condition is to be kept, it should be therefore moved only to the midpoint refresh case. I do not insist on it to be kept, because it handles more or less misconfiguration of those two options. I will leave decision of the author of this patch.